docs/metrics/average-precision.md

```diff
@@ -0,0 +1,250 @@
+# Average Precision
```

Comment:
Add search exclusion? Looks like the plan is to land this with the temp diagrams and follow up with the production diagrams + links to this guide on the sidebar and index page.